Product/Production/Services/CORE_X12DocumentSubmissionCore/src/main/java/gov/hhs/fha/nhinc/corex12/docsubmission/audit/COREX12BatchAuditTransforms.java

```diff
@@ -0,0 +1,297 @@
+            CoreX12AuditDataTransformConstants.EVENT_ID_CODE_SYS_NAME_X12, CoreX12AuditDataTransformConstants.EVENT_ID_DISPLAY_NAME_X12REALTIME));
+        auditMsg.setEventIdentification(oEventIdentificationType);
+
+        AuditMessageType.ActiveParticipant participantSource = oCoreRealTimeTransforms.getActiveParticipantSource(isRequesting);
+        AuditMessageType.ActiveParticipant participantDestination = oCoreRealTimeTransforms.getActiveParticipantDestination(target);
+        auditMsg.getActiveParticipant().add(participantSource);
+        auditMsg.getActiveParticipant().add(participantDestination);
+
+        /* Assign ParticipationObjectIdentification */
+        ParticipantObjectIdentificationType participantObject = oCoreRealTimeTransforms.getParticipantObjectIdentificationType(response);
+        auditMsg.getParticipantObjectIdentification().add(participantObject);
+
+        //find the requesting or responding gateway
+        String messageDirection = isRequesting ? NhincConstants.AUDIT_LOG_OUTBOUND_DIRECTION : NhincConstants.AUDIT_LOG_INBOUND_DIRECTION;
+
+        /* Create the AuditSourceIdentifierType object */
+        String communityId = oXDR.getMessageCommunityIdFromRequest(assertion, target, messageDirection, _interface);
+        AuditSourceIdentificationType auditSource = oCoreRealTimeTransforms.getAuditSourceIdentificationType();
+        auditMsg.getAuditSourceIdentification().add(auditSource);
+
+        result.setAuditMessage(auditMsg);
+        result.setDirection(direction);
+        result.setInterface(_interface);
+        LOG.trace("End COREX12RealtimeTransforms -> transformBatchResponseToAuditMsg() -- NHIN");
+        return result;
+    }
+
+    private boolean areRequiredCOREX12BatchFiledsNull(COREEnvelopeBatchSubmission msg, AssertionType assertion) {
+        if (oXDR.areRequiredUserTypeFieldsNull(assertion)) {
+            LOG.error("One of more UserInfo fields from the Assertion object were null.");
+            return true;
+        }
+
+        if (areRequiredRequestFieldsNull(msg)) {
+            LOG.error("One or more fields in CORE X12 Batch Request object were null.");
+            return true;
+        }
+        return false;
+    }
+
+    private boolean areRequiredCOREX12BatchFiledsNull(COREEnvelopeBatchSubmissionResponse response, AssertionType assertion) {
+        if (oXDR.areRequiredUserTypeFieldsNull(assertion)) {
+            LOG.error("One of more UserInfo fields from the Assertion object were null.");
+            return true;
+        }
+
+        if (areRequiredRequestFieldsNull(response)) {
+            LOG.error("One or more fields in CORE X12 Batch Response object were null.");
+            return true;
+        }
+        return false;
+    }
+
+    private boolean areRequiredRequestFieldsNull(COREEnvelopeBatchSubmission msg) {
```

Comment:
There are a LOT of places in this PR where you see "if x, return true; if y, return true;" -- in larger functions, this creates many exit points, and makes JPDA debugging much more difficult.  Consider altering the flow in instances like this to use multiple "else if" statements that set a boolean instead of returning.  This will not impact performance, but will make things cleaning.